libcudacxx/include/cuda/std/__complex/nvfp16.h

```diff
@@ -87,6 +87,15 @@ class _LIBCUDACXX_TEMPLATE_VIS _CCCL_ALIGNAS(alignof(__half2)) complex<__half>
       : __repr_(static_cast<value_type>(__c.real()), static_cast<value_type>(__c.imag()))
   {}
 
+  template <class _Up,
+            __enable_if_t<!__complex_can_implicitly_construct<value_type, _Up>::value, int> = 0,
+            __enable_if_t<!_CCCL_TRAIT(is_constructible, value_type, _Up)
+                            && (_CCCL_TRAIT(is_same, _Up, float) || _CCCL_TRAIT(is_same, _Up, double)),
+                          int>                                                              = 0>
+  _LIBCUDACXX_INLINE_VISIBILITY explicit complex(const complex<_Up>& __c)
+      : __repr_(__float2half(__c.real()), __float2half(__c.imag()))
+  {}
+
```

Comment:
I believe we should not use a template here, but explicit constructors:
```suggestion
  _LIBCUDACXX_INLINE_VISIBILITY explicit complex(const complex<float>& __c)
      : __repr_(__float2half(__c.real()), __float2half(__c.imag()))
  {}
  _LIBCUDACXX_INLINE_VISIBILITY explicit complex(const complex<double>& __c)
      : __repr_(__double2half(__c.real()), __double2half(__c.imag()))
  {}

```

Reply:
Done